From 0eff67e17d7a38ed985e799e08a416c87022ed5e Mon Sep 17 00:00:00 2001
From: Vasu Kantubukta <vasu.kantubukta@lge.com>
Date: Wed, 19 Dec 2018 21:02:59 +0530
Subject: [PATCH] Fix for wifi network switching and unable to connect to open

:Release Notes:
Fix for selected wifi network switching and unable to connect
popup.

:Detailed Notes:
Whenever switching to different network, interface state will
be triggered with state DISCONNECT which disturbing the pending
wifi network connection status. Connecting to open network is a
side effect of the same. This patch applies a fix switch to
selected network.

:Testing Performed:
Build is successfully done.
Verified the image.

:QA Notes:

:Issues Addressed:
[PLAT-57861] [webOS OSE] WiFi Connection : WiFi Network does not
             switch to the selected network.
[PLAT-65726] [webOS OSE] : "Unable to connect to the network" Popup
             appears instead of Password prompt.

Upstream-Status: Inappropriate [webOS specific]
---
 plugins/wifi.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/plugins/wifi.c b/plugins/wifi.c
index 31a2a396..296ac113 100644
--- a/plugins/wifi.c
+++ b/plugins/wifi.c
@@ -2973,7 +2973,7 @@ static void interface_state(GSupplicantInterface *interface)
 		}
 		wifi->disconnecting = false;
 
-		if (perform_deferred_connect_after_disconnect(wifi) == FALSE)
+		if (!deferred_wps_start(wifi))
 		{
 			/* Set connected to false to allow autoscan to start. */
 			wifi->connected = FALSE;
